### PR TITLE
Repo configuration

### PR DESCRIPTION
**Pull request by nomaxg**
Nix, CODEOWNERS, and fork sync Github action

### CODE REVIEW

File: .github/workflows/fork-sync.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  sync:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: tgymnich/fork-sync@v1.8
+        with:
+          owner: ethereum-optimism
+          repo: optimism
+          base: op-espresso-integration # our default branch
```
Comment (jbearer) [inline .github/workflows/fork-sync.yml]: This is the name of the fork, not our default branch. Either this value or the comment should be changed.

File: .github/workflows/fork-sync.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+name: Sync Fork
+
+on:
+  schedule:
+    - cron: '0 * * * *' # every hour
```
Comment (jbearer) [inline .github/workflows/fork-sync.yml]: Every hour seems excessive. Maybe once a day or once a week.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+/nix/store/j9wh97mfzimd9rq2493y15i9m5prkrhw-pre-commit-config.json
```
Comment (jbearer) [inline .pre-commit-config.yaml]: Where did this come from? This is not yaml

Comment (jbearer) [reply .pre-commit-config.yaml]: This looks like a symlink that somehow got committed as a regular file?

**Review by jbearer — CHANGES_REQUESTED** [review]

**Comment by sveitser** [comment]
I think this github action is not ideal because it fails if there are no changes and it fails if there is a PR already (instead of updating it). 

I think it's worth trying https://github.com/wei/pull and see if that has a more sane behaviour.

**Comment by nomaxg** [comment]
> I think this github action is not ideal because it fails if there are no changes and it fails if there is a PR already (instead of updating it).
> 
> I think it's worth trying https://github.com/wei/pull and see if that has a more sane behaviour.

Ok, let's try that instead, I'll delete fork-sync for now

**Review by jbearer — APPROVED** [review]

**Pull request merged**